adds syntax highlights to ExDoc buffers when `!alchemist#ansi_enabled()`

c-brenn opened the pull request:
Made this change locally a few days ago - committing and opening a PR in case others would find it useful.

This change adds a simple syntax file for ExDoc buffers and enables it when the user
opens `ExDoc` and does not have `AnsiExec`. Previously, the ExDoc buffer relied on `markdown` syntax highlighting.

It covers some basic cases - code blocks, section headers etc - but I'm sure there are some edge cases it misses.

Below is a screenshot with an `ExDoc` buffer on the left and `iex` on the right.

![screen shot 2017-03-19 at 19 55 47](https://cloud.githubusercontent.com/assets/8920322/24084252/7ed470c2-0cde-11e7-951d-d4a7d0e847e1.png)

c-brenn committed:
adds syntax highlights to ExDoc buffers when `!alchemist#ansi_enabled()`

Adds a simple syntax file for ExDoc buffers and enables it when the user
opens `ExDoc` and does not have `AnsiExec`

slashmili commented:
Looks great! Thanks

I'll also try it out in next few days we can merge it next week

c-brenn commented:
Awesome! Let me know if you notice any incorrect highlighting and I'll fix it up.

c-brenn committed:
use colours as opposed to highlight groups

Different colorschemes will hightlight groups differently. This changes
the exdoc syntax file to use hard coded colours - so everyone's vim
looks like iex.

The colours are a first approximation - easily changed.

slashmili commented:
So far Looks good!

Thanks @c-brenn for the contribution 🎉
